### ISSUE TITLE
make animations clearer

### ISSUE THREAD

**Issue by umlaeute** [pd-externals/tipoftheday-plugin#28]
> as for a common format: i don't really know, but wouldn't overthink this now
[...]
> for keyboard shortcuts i was thinking about using animated GIFs to illustrate their use.

with my #13 (and your #22, which follows the spirit) i now see that I'm not totally happy with the current keyboard-shortcut-animations, as they don't really convey the chain of cause and effect.

i guess the most important thing is to establish
1. how it looked **before**
2. which **action** needs to be performed
3. how it looks **after**wards

allowing the user to distinguish between *before* and *after* probably means, that they should be aware of the *start* of the animation.
we probably could automatically insert one (or more) empty frame(s) at the end of the animation, but i think this will mostly get in the way of nice animations.

anyhow: maybe somebody with more knowledge could suggest a template how such animated GIFs should look like...

i'm having a hard time finding something useful on the internet though:
![foo](https://media.giphy.com/media/l0MYtKJXuLlLgCxby/giphy.gif)

_Originally posted by @umlaeute in https://github.com/pd-externals/tipoftheday-plugin/issues/10#issuecomment-1153534835_

**Comment by umlaeute** [comment]
so as a first remedy, i've disabled the auto-looping of animations.

this way, an animation has a clear beginning and an end (though less clear).

in order to restart the animation, the user must now click anywhere in the tip (for whatever reasons i couldn't add a tag to the image itself, so it was not possible to bind the click only to the image).

the question is: how to tell people that they can restart the animation? adding some text to each tip (with an image) seems like a lot of noise (once you got the idea)... add another tip?

**Comment by aaaaalbert** [comment]
> i guess the most important thing is to establish
> 
> 1. how it looked before
> 2. which action needs to be performed
> 3. how it looks afterwards

I agree. My attempt in #22 to show the start / stop of the GIF was to move the mouse cursor into the window a few moments after the video starts, and move it out of the window after showing the interaction. The GIF took a few takes before I felt it showed the sequence of pressing <kbd>Ctrl</kbd>, clicking the object, releasing the key etc. slowly enough to be discernible. (Other issues remain, e.g. the number box gets out of focus as the GIF loops.)

> i've disabled the auto-looping of animations.

I think looping the GIFs is better than not looping. Memes made us grow to expect this.

Random thought mode on:
* Some common visual indicator of the start and stop would make sense. Could be a simple, quick fade from/to black.
* On the other hand, for many cases a GIF almost feels like too much. Like, *just Ctrl-click that number box a few times, OK?* Not that hard!
* Or open a new can of worms and include / allow to open additional small "tip patches" where you can try out the tip right away, probably with more commentary. (Inspired by a colleague who is working on a Pd beginner's manual in the form of patches, which in turn is inspired by prior work in [a Pd messages tutorial](https://github.com/fdch/pd-messages), [a GUI tutorial](https://puredata.info/community/conventions/convention04/lectures/tk-barknecht/tut.tgz/view), and [a tool for making multiple connections](https://forum.pdpatchrepo.info/uploads/files/1639078420437-connector.zip).)

**Comment by umlaeute** [comment]
i like tip-patches (as long as they follow the spirit and only teach a single thing)

please open a feature request ;-)


as for looping: you're probably right. although i quite like the feel of the current non-looping GIFs (it makes it much easier to follow, as the i can focus my attention on the entire sequence)
